apps/web/src/components/editor/action-bar.tsx

```diff
@@ -314,6 +314,7 @@ function TabStrip() {
               isActive={session.id === activeSessionId}
               isPinned={!!session.pinned}
               isLocked={isLockedSession(session)}
+              isOnly={sessions.length === 1}
```

Comment:
Instead of all this wizardry, let's just move the listeners from the parent `Flex` to the sibling `Flex`